client/internal/wsreceiver_test.go

```diff
@@ -217,3 +218,62 @@ func TestReceiverLoopStop(t *testing.T) {
 		return receiverLoopStopped.Load()
 	}, 2*time.Second, 100*time.Millisecond, "ReceiverLoop should stop when context is cancelled")
 }
+
+func TestWSPackageUpdatesInParallel(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	var messages atomic.Int32
+	var mux sync.Mutex
+	localPackageState := NewInMemPackagesStore()
```

Comment:
Is there a way we can mock InMemPackageStore to block on some operation so that we can wait for the second `Sync` operation to be called here in the test? Without that we do not necessarily test the correctness of the mutexes and only rely on race detector to detect problem (which is not guaranteed).

Reply:
Hey, sorry for taking so long to get back to you Tigran. 👋 

I've pushed a new commit in https://github.com/open-telemetry/opamp-go/pull/289/commits/273d37b095b2de2c6fa87326ca3d303e8e944e5a that tries to do that.

For the first call, it blocks on a channel, and changes the condition so that both Sync() calls have returned _before_ releasing the block and letting the two Sync calls complete in sequence. The race detector would catch this more readily I think. Is that along what you had in mind?